Lib/test/test_typing.py

```diff
@@ -3132,6 +3132,20 @@ def x(self): ...
         with self.assertRaisesRegex(TypeError, only_classes_allowed):
             issubclass(1, BadPG)
 
+    def test_lazy_evaluation_with_subprotocols(self):
+        @runtime_checkable
+        class Base(Protocol):
+            x: int
+
+        class Child(Base, Protocol):
```

Comment:
can't remember if this is necessary or not if the base class has `runtime_checkable` set, but calling `isinstance()` against a protocol looks weird to me if the protocol is not decorated with `@runtime_checkable` 😄

```suggestion
        @runtime_checkable
        class Child(Base, Protocol):
```

Reply:
Good catch. It's actually weird that this passed, shouldn't we raise an error if you isinstance() on a non-checkable protocol?

Reply:
I think so. But this seems like longstanding behaviour, dating back to Python 3.8:

```pycon
Python 3.8.18 (default, Feb 15 2024, 19:36:58) 
[Clang 15.0.0 (clang-1500.1.0.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import runtime_checkable, Protocol
>>> @runtime_checkable
... class Foo(Protocol):
...     x: int
... 
>>> class Bar(Foo, Protocol):
...     y: str
... 
>>> isinstance(object(), Bar)
False
```

So I don't think we could change this without a deprecation period, unfortunately. Someone somewhere is probably accidentally relying on it by now.

Reply:
I see, it's because we do `getattr(cls, '_is_runtime_protocol', False)` and that picks up the attribute from the base class. I feel like this is a bug so I'll open an issue about it, but agree that compatibility may mean we shouldn't change it.